pkg/cloud/services/secretsmanager/secret_test.go

```diff
@@ -0,0 +1,291 @@
+	}
+}
+
+func TestService_Delete(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tests := []struct {
+		name        string
+		secretCount int32
+		expect      func(m *mock_secretsmanageriface.MockSecretsManagerAPIMockRecorder)
+		check       func(*WithT, error)
+	}{
+		{
+			name:        "Should not call AWS when secret count has zero value",
+			secretCount: 0,
+			check: func(g *WithT, err error) {
+				g.Expect(err).NotTo(HaveOccurred())
+			},
+		},
+		{
+			name:        "Should not return error when delete is successful",
+			secretCount: 1,
+			expect: func(m *mock_secretsmanageriface.MockSecretsManagerAPIMockRecorder) {
+				m.DeleteSecret(gomock.Eq(&secretsmanager.DeleteSecretInput{
+					SecretId:                   aws.String("prefix-0"),
+					ForceDeleteWithoutRecovery: aws.Bool(true),
+				})).Return(&secretsmanager.DeleteSecretOutput{}, nil)
+			},
+			check: func(g *WithT, err error) {
+				g.Expect(err).NotTo(HaveOccurred())
+			},
+		},
+		{
+			name:        "Should return all errors except not found errors",
+			secretCount: 3,
+			expect: func(m *mock_secretsmanageriface.MockSecretsManagerAPIMockRecorder) {
+				m.DeleteSecret(gomock.Eq(&secretsmanager.DeleteSecretInput{
+					SecretId:                   aws.String("prefix-0"),
+					ForceDeleteWithoutRecovery: aws.Bool(true),
+				})).Return(nil, awserrors.NewFailedDependency("failed dependency"))
+				m.DeleteSecret(gomock.Eq(&secretsmanager.DeleteSecretInput{
+					SecretId:                   aws.String("prefix-1"),
+					ForceDeleteWithoutRecovery: aws.Bool(true),
+				})).Return(nil, awserrors.NewNotFound("not found"))
+				m.DeleteSecret(gomock.Eq(&secretsmanager.DeleteSecretInput{
+					SecretId:                   aws.String("prefix-2"),
+					ForceDeleteWithoutRecovery: aws.Bool(true),
+				})).Return(nil, awserrors.NewConflict("new conflict"))
+			},
+			check: func(g *WithT, err error) {
+				g.Expect(err).ToNot(BeNil())
+				g.Expect((err.Error())).To(Equal("[failed dependency, new conflict]"))
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			client := setupClient()
+			clusterScope, err := getClusterScope(client)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			secretManagerClientMock := mock_secretsmanageriface.NewMockSecretsManagerAPI(mockCtrl)
+			if tt.expect != nil {
+				tt.expect(secretManagerClientMock.EXPECT())
+			}
+			s := NewService(clusterScope)
+			s.SecretsManagerClient = secretManagerClientMock
+			ms, err := getMachineScope(client, clusterScope)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			ms.SetSecretPrefix("prefix")
+			ms.SetSecretCount(tt.secretCount)
+			err = s.Delete(ms)
+			tt.check(g, err)
+		})
+	}
+}
+
+func setupClient() client.Client {
+	scheme := runtime.NewScheme()
+	_ = infrav1.AddToScheme(scheme)
+	return fake.NewClientBuilder().WithScheme(scheme).Build()
+}
+
+func getClusterScope(client client.Client) (*scope.ClusterScope, error) {
```

Comment:
nit. Duplicated code with other services.

Reply:
As suggested in other comment, we can do this after other related PRs are also merged.